File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/ui/view/FMEditView.java
package de.dimedis.mobileentry.ui.view;

import static de.dimedis.mobileentry.util.SoftKeyboard.showKeyboard;

import android.content.Context;
import android.text.Editable;
import android.util.AttributeSet;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import de.dimedis.mobileentry.R;

public class FMEditView extends LinearLayout {
    EditText mEditText;
    ImageButton mImageButton;

    public FMEditView(Context context) {
        super(context, null);
    }

    public FMEditView(Context context, AttributeSet attrs) {
        super(context, attrs, 0);
        init(context, attrs);
    }

    void init(Context context, AttributeSet attrs) {
        inflate(getContext(), R.layout.view_custedit, this);
        mEditText = findViewById(R.id.editText);
        mImageButton = findViewById(R.id.imageButton);
        mImageButton.setOnClickListener(v -> softKeyboard());
    }

    public void setOnEditorActionListener(TextView.OnEditorActionListener l) {
        mEditText.setOnEditorActionListener(l);
    }

    public Editable getText() {
        return mEditText.getText();
    }

    void softKeyboard() {
        mEditText.requestFocus();
        showKeyboard(mEditText);
    }

    @Override
    protected void onDetachedFromWindow() {
        super.onDetachedFromWindow();
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/ui/view/FMEditView.kt
package de.dimedis.mobileentry.ui.view

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView.OnEditorActionListener
import android.text.Editable
import android.view.View
import de.dimedis.mobileentry.R
import de.dimedis.mobileentry.util.SoftKeyboard

class FMEditView : LinearLayout {
    var mEditText: EditText? = null
    var mImageButton: ImageButton? = null

    constructor(context: Context?) : super(context, null) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs, 0) {
        init(context, attrs)
    }

    fun init(context: Context?, attrs: AttributeSet?) {
        inflate(getContext(), R.layout.view_custedit, this)
        mEditText = findViewById(R.id.editText)
        mImageButton = findViewById(R.id.imageButton)
        mImageButton?.setOnClickListener(OnClickListener {
            softKeyboard() })
    }

    fun setOnEditorActionListener(l: OnEditorActionListener?) {
        mEditText!!.setOnEditorActionListener(l)
    }

    fun getText(): Editable {
        return mEditText!!.text
    }

    fun softKeyboard() {
        mEditText!!.requestFocus()
        SoftKeyboard.showKeyboard(mEditText)
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
    }
}